### PR TITLE
update pmd, replace deprecations, set RELEASE 12 version

### PR DESCRIPTION
**Pull request by KroArtem**
### Context
* Set `SourceVersion.RELEASE_12` instead of `SourceVersion.RELEASE_8`
* Updated PMD and checkstyle (suppressed one violation)
* Replaced deprecations

#### Checklist
- [x] [Sign Allure CLA][cla]
- [ ] ~Provide unit tests~

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

**Comment by KroArtem** [comment]
Ok, seems like I need some time to link commits to current user and to investigate what version is used to build on Jenkins

**Comment by baev** [comment]
on Jenkins we are using java 8. I doubt we can change it to 12 any time soon as Jenkins has no support for Java 12 yet

**Comment by KroArtem** [comment]
That's a pity. I'll revert to `SourceVersion.RELEASE_8` then. The reason was a warning that happens in our project:
Screen Shot 2019-06-17 at 16 48 45

**Comment by KroArtem** [comment]
@baev , I believe everything is ok now. I've removed type `AutoCloseable` from PMD rule as it was added in 6.16 version https://pmd.github.io/pmd-6.16.0/pmd_rules_java_errorprone.html#closeresource

File: allure-jax-rs/src/main/java/io/qameta/allure/jaxrs/AllureJaxRs.java [hunk 1]
```diff
@@ -116,6 +116,8 @@ private String getBody(final ClientResponseContext responseContext) throws IOExc
             }
             responseContext.setEntityStream(new ByteArrayInputStream(result.toByteArray()));
             return result.toString(StandardCharsets.UTF_8.toString());
+        } finally {
+            stream.close();
```
Comment (baev) [inline allure-jax-rs/src/main/java/io/qameta/allure/jaxrs/AllureJaxRs.java]: use try-with-resources instead

Comment (KroArtem) [reply allure-jax-rs/src/main/java/io/qameta/allure/jaxrs/AllureJaxRs.java]: done

**Comment by baev** [comment]
In order to make cla bot happy, you need to get rid of commits signed as `artem.krosheninnikov` or link them with you'r github account

**Comment by KroArtem** [comment]
As github says,

`
Unrecognized author (no email address) If you see this message without an email address, it means you used a generic email address that can't be added to your email settings. You will need to set your commit email address in Git, then add the new address to your GitHub email settings to link your future commits. Old commits will not be linked.
`

https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user

I've fixed it for commits made since today but cannot make it for previous one

**Comment by baev** [comment]
```bash
git reset --soft HEAD~4
git commit -am "new cool commit message"
git push -f 
```

**Comment by KroArtem** [comment]
We do not use resetting changes that were pushed to remote in our project, but if that's ok here, why not. Sorry for the delay.

**Comment by baev** [comment]
@KroArtem thanks!

**Pull request merged**